content/ui-patterns/degraded-experiences.mdx

```diff
@@ -0,0 +1,518 @@
+
+Explain what's wrong and, if possible, link to a page with more detailed information. For example, if there's a database outage, we could link to the [GitHub status page](https://www.githubstatus.com/).
+
+In addition to a global banner, we should inform users about availability issues in context. The following guidelines discuss how to handle outages in the context of the affected UI.
+
+</div>
+
+<img
+  width="456"
+  alt="Image of a page with a banner at the top explaining an error"
+  src="https://github.com/primer/design/assets/2313998/b4a577c5-1243-435f-8dfc-c2a9d292530e"
+/>
+
+</Box>
+
+## Degraded page content
+
+If part of the UI cannot be rendered or would be rendered without critical information, default to not rendering it at all.
+
+Before making a decision about how to handle UI with unavailable content, check if it already has guidelines on handling errors or empty states.
+
+### Removing UI
+
+You can remove affected UI if it's not critical to core workflows and it wouldn't be confusing the render the rest of the page without that UI. For example: it's ok to hide the reactions button.
+
+Examples of UI that might be disorienting to remove:
+
+- The comment box on issues and pull requests
+- The "New issue" button on the issues page
+- Submit buttons on forms
+
+### Replacing UI with a message
+
+<img
+  width="960"
+  alt="Two images: 1. Assignees removed from issue sidebar; 2. Assignee column in a Memex table with 'data could not be loaded' message in cells"
+  src="https://github.com/primer/react/assets/2313998/a2da5f80-1e32-423c-85b8-ca77f936d31c"
+/>
+
+We don't want users to think they've suffered data loss. If we know a user created something, don't show a generic "empty state". It's better to explain that it's unavailable or remove it entirely.
+
+#### Inline content and smaller elements
+
+Smaller parts of the UI that cannot be accurately rendered but are too important to exclude entirely can often be replaced with a short error message.
+
+<img
+  width="960"
+  alt="Two images: 1. Image of Memex table with populated content cells; 2. Image of Memex table with missing content message in cells"
+  src="https://github.com/primer/design/assets/2313998/fffb1ff8-33cc-405a-9149-44d67a51b1d0"
+/>
+
+{/* <!-- TODO: Try and come up with more specific guidance and examples for how to use `fg.warning` --> */}
+
+By default, replace the affected content with an error message. Show a warning icon before the message to help differentiate it from non-degraded content. The message may be colored with `fg.warning` to draw more attention to it.
+
+Be mindful that rendering too many error messages on the page in `fg.warning` could be jarring and make the page feel broken instead of degraded.
```

Comment:
> There's no single entity which knows about all the individual elements on the page and which could control how many error messages can be shown.

This is good context to consider. I'd rather we accidentally show multiple error messages than not render an important error message.

---

> Also, if we could somehow determine that there will be lots (more than 5?) of error messages, what should be done then? Should 5 be shown and no more? 

5 seems like a reasonable suggested limit. It really depends on what else is on the page. I think we just want to avoid overwhelming a user with a bunch of error messages.

---

> Or should none be shown and just the warning banner at the top is displayed? 🤔

In most cases, content won't need to be replaced with an error message except when it would be distressing or disorienting to see an empty space. For example, I might think my repositories were deleted if I'm trying to view a list of my repositories on a page like https://github.com/primer/ and there's nothing there.

Reply:
>> Also, if we could somehow determine that there will be lots (more than 5?) of error messages, what should be done then? Should 5 be shown and no more?
>
> 5 seems like a reasonable suggested limit. It really depends on what else is on the page. I think we just want to avoid overwhelming a user with a bunch of error messages.

Ah, I apologize. My question was mostly rhetorical since, to me, it seemed like any specific number could be "right" or could be "wrong" since it all depends on the contents of the page. E.g. why is 5 good and 6 is bad, and such.

What you wrote "It really depends on what else is on the page." is exactly the problem -- there's no way one nested part of the page can determine what other nested parts are on the page. Each part lives in it's own little world isolated of other parts. :shrug: